client/components/auto-direction/index.jsx

```diff
@@ -37,9 +77,18 @@ const getContent = ( reactElement ) => {
 	// This child has it's content set to external HTML
 	if ( typeof props.dangerouslySetInnerHTML === 'object' ) {
 		// Strip tags because we're only interested in the text, not markup
-		return props.dangerouslySetInnerHTML.__html
-			? stripHTML( props.dangerouslySetInnerHTML.__html.substring( 0, Math.floor( MAX_LENGTH_OF_TEXT_TO_EXAMINE * 1.25 ) ) )
-			: '';
+		// We examine from the first position without tags of the string, so we won't get an empty text
+		// because we might get only tags in the beginning
+		const html = props.dangerouslySetInnerHTML.__html;
+		if ( ! html ) {
+			return '';
+		}
+
+		const examinedLength = Math.floor( MAX_LENGTH_OF_TEXT_TO_EXAMINE * 1.25 );
+		const taglessIndex = getTaglessIndex( html );
+		const startIndex = taglessIndex + examinedLength < html.length ? taglessIndex : 0;
```

Comment:
1. It was a left over from the previous code, I changed it to just use MAX_LENGTH_OF_TEXT_TO_EXAMINE,
the 1.25 was a hack to compensate on the tag. 
2. The purpose is either start from the first index without tags if the length of the text allows, or from the begining, as the comment at line 80 says.
3. It can be great, I might get to it sometime